cloudplatform/tenant/src/main/java/com/sap/cloud/sdk/cloudplatform/tenant/DefaultTenant.java

```diff
@@ -6,15 +6,39 @@
 
 import javax.annotation.Nonnull;
 
-import lombok.Data;
+import lombok.EqualsAndHashCode;
 import lombok.Getter;
 import lombok.RequiredArgsConstructor;
+import lombok.ToString;
 
+/**
+ * Implementation of {@link Tenant} which can be used on SAP Business Technology Platform Cloud Foundry.
+ */
+@EqualsAndHashCode
+@ToString
 @RequiredArgsConstructor
-@Data
-public class DefaultTenant implements Tenant
+public class DefaultTenant implements TenantWithSubdomain
 {
+    /**
+     * The identifier of the tenant.
+     */
     @Getter
     @Nonnull
     private final String tenantId;
+
+    /**
+     * The subdomain of the tenant.
+     */
+    @EqualsAndHashCode.Exclude
+    @Getter
+    @Nonnull
+    private final String subdomain;
+
+    /**
+     * Creates a mocked {@link DefaultTenant} with an empty tenant identifier.
+     */
+    public DefaultTenant()
+    {
+        this("", "");
+    }
```

Comment:
(Major)

I have some problems with that change:
* Please don't expose a tenant constructor without tenant id. Leaving a note in JavaDoc hinting for "mocking" would not be enough for users to discourage the constructor usage.
* Please don't breaking change the default constructor. It should always be possible to create a default tenant just with id.
* I assume DwC is a requirement to use `TenantWithSubdomain`. If that's the case, and we can't guarantee a subdomain available, then I would say, let's throw an exception instead of returning empty string. `""` could lead to very confusing error messages in the long run, when someone tries to replace a hostname in their URL.

```suggestion
    @EqualsAndHashCode.Exclude
    @Nullable
    private final String subdomain;
    
    /**
     * Creates a {@link DefaultTenant} without subdomain.
     */
    public DefaultTenant( @Nonnull final String tenantId)
    {
        this(tenantId, null);
    }

    @Nonnull
    @Override
    public String getSubdomain() {
        return Objects.requireNonNull(subdomain, "Subdomain is not available for this tenant.");
    }
```

Reply:
> Please don't expose a tenant constructor without tenant id. Leaving a note in JavaDoc hinting for "mocking" would not be enough for users to discourage the constructor usage.

Interesting, do you know what the intended usage of this constructor was when it was part of the `ScpCfTenant`? Just out of curiosity. 

But I see your point with needing a tenantId-only constructor. Will commit. :)